x-pack/plugins/enterprise_search/public/applications/shared/layout/page_template.tsx

```diff
@@ -34,6 +36,7 @@ import './page_template.scss';
  */
 
 export type PageTemplateProps = KibanaPageTemplateProps & {
+  access?: ProductAccess;
```

Comment:
suggestion: I would make this a mandatory instead of an optional prop, so that we force developers to pass this to every implementation of the PageTemplate. Otherwise we will miss it at some point in the future, and that'll lead to inconsistent implementation of the access.

Reply:
I agree that future instances of `PageTemplate` should be passed `access`. We made this optional to avoid having to make changes to instances of `AppSearchPageTemplate` ([like this](https://github.com/elastic/kibana/blob/6693ef371f887eca639b09c4c9b15701b4ebabd4/x-pack/plugins/enterprise_search/public/applications/app_search/components/source_engines/source_engines.tsx#L35)) and `WorkplaceSearchPageTemplate` ([like this](https://github.com/elastic/kibana/blob/6693ef371f887eca639b09c4c9b15701b4ebabd4/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/organization_sources.tsx#L39)). Both of those products avoid the problem entirely by not showing links to each other, and so don't need `access`.

I suppose we could change them to use a different interface than `PageTemplateProps`...would that be preferable?

Reply:
Yeah, I think splitting off those two templates to use a different set of props, or splitting off *this* template to use a different set of props would be preferable. Because really, we do want this to be a mandatory prop going forward, I think.